tests/integration/security/mtls_healthcheck_test.go

```diff
@@ -12,55 +12,77 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-// Package healthcheck contains a test to support kubernetes app health check when mTLS is turned on.
-// https://github.com/istio/istio/issues/9150.
-package healthcheck
+package security
 
 import (
+	"fmt"
+	"strconv"
 	"testing"
+	"time"
 
+	"istio.io/istio/pkg/config/protocol"
 	"istio.io/istio/pkg/test/framework"
 	"istio.io/istio/pkg/test/framework/components/echo"
+
 	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
 	"istio.io/istio/pkg/test/framework/components/environment"
 	"istio.io/istio/pkg/test/framework/components/namespace"
 )
 
 // TestMtlsHealthCheck verifies Kubernetes HTTP health check can work when mTLS
-// is enabled.
+// is enabled, https://github.com/istio/istio/issues/9150.
 // Currently this test can only pass on Prow with a real GKE cluster, and fail
 // on Minikube. For more details, see https://github.com/istio/istio/issues/12754.
 func TestMtlsHealthCheck(t *testing.T) {
 	framework.NewTest(t).
 		RequiresEnvironment(environment.Kube).
 		Run(func(ctx framework.TestContext) {
+			ns := namespace.ClaimOrFail(t, ctx, "healthcheck")
+			runHealthCheckDeployment(t, ctx, ns, "ok", true, true)
+			runHealthCheckDeployment(t, ctx, ns, "fail", false, false)
```

Comment:
Should this be cleaner to run as separate test cases?

Reply:
done.